activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -726,23 +728,41 @@ def default_sequence_name(table_name, pk = nil) #:nodoc:
         default_seq || "#{table_name}_#{pk || default_pk || 'id'}_seq"
       end
 
+      #
+      # Monkey-patch the refused Rails 4.2 patch at https://github.com/rails/rails/pull/31330
+      # Changed the module/class hierarchy to work with Rails 3.2
+      # Based on solution for Rails 4.2 https://github.com/rails/rails/issues/28780#issuecomment-354868174
+      #
+      # Updates sequence logic to support PostgreSQL 10.
+      #
```

Comment:
see also https://github.com/Genius/Rap-Genius/blob/master/config/initializers/active_record_extensions.rb#L583-L609

Reply:
This has been pulled out into rails2_ruby2.